### PR TITLE
Fix for sdk25

### PR DESCRIPTION
**Pull request by rjaros87**

### CODE REVIEW

**Comment by rjaros87** [comment]
@brantone , @gildegoma please review. Fix for https://github.com/gildegoma/chef-android-sdk/issues/34

**Comment by Brantone** [comment]
Seems like lots of other peripheral changes (ex: remove tailor from Rakefile), and bunch of white space changes, which make it somewhat cumbersome to isolate actual changes to SDK25.
Possible to split out the actual changes for  gildegoma#34 SDK25 and extras? Or even maybe few extra comments for other un-referenced chanages?

File: Rakefile [hunk 1]
```diff
@@ -2,40 +2,13 @@
...
-require 'tailor/rake_task'
 require 'rubocop/rake_task'
 require 'rspec/core/rake_task'
 
 # TODO: add chefspec
-task default: [:tailor, :rubocop, :foodcritic, :knife, :serverspec]
+task default: [:rubocop, :foodcritic, :knife, :serverspec]
 
 desc 'Lint Ruby code'
-task :tailor do
```
Comment (Brantone) [inline Rakefile:13]: Why removal?

Comment (rjaros87) [reply Rakefile:13]: Rubocop checks the the same and even more.

File: .kitchen.yml [hunk 2]
```diff
@@ -8,88 +8,97 @@ provisioner:
...
-      group: vagrant
-    components:
-      - build-tools-22.1.0
-      - android-22
-      - sysimg-22
-    java:
-      jdk_version: 7
-
-- name: default
+- name: legacy
```
Comment (rjaros87) [inline .kitchen.yml:85]: Suite to check the backward compatibility

File: .kitchen.yml [hunk 3]
```diff
@@ -8,88 +8,97 @@ provisioner:
...
+        linux: f2bb546534d16e2004665257ee530060338c684adad14a49cd4bbde08098d8a4
+        mac_os_x: a1cb0c9b2036d597e18986de1bd43918b6113373ff3bf6fdb81b6cce4ec3efd7
+        windows: 1f9f22271fe55bf0bf10b60556ed1a5fa2f2ca672fc20d35478325104941cb05
       components:
         - platform-tools
         - build-tools-22.0.1
         - android-22
         - sys-img-armeabi-v7a-android-22
+
+- name: default
```
Comment (rjaros87) [inline .kitchen.yml:103]: Suite to check the _latest_ SDK and default recipe without modification attributes `android-sdk`

File: attributes/default.rb [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
-default['android-sdk']['checksum']['windows']       = '1f9f22271fe55bf0bf10b60556ed1a5fa2f2ca672fc20d35478325104941cb05'
-default['android-sdk']['download_url']['linux']     = "http://dl.google.com/android/android-sdk_r#{node['android-sdk']['version']}-linux.tgz"
-default['android-sdk']['download_url']['mac_os_x']  = "https://dl.google.com/android/android-sdk_r#{node['android-sdk']['version']}-macosx.zip"
-default['android-sdk']['download_url']['windows']   = "https://dl.google.com/android/android-sdk_r#{node['android-sdk']['version']}-windows.zip"
+default['android-sdk']['version']                     = '25.2.3'
+default['android-sdk']['checksum']['linux']           = '1b35bcb94e9a686dff6460c8bca903aa0281c6696001067f34ec00093145b560'
+default['android-sdk']['checksum']['mac_os_x']        = '593544d4ca7ab162705d0032fb0c0c88e75bd0f42412d09a1e8daa3394681dc6'
+default['android-sdk']['checksum']['windows']         = '23d5686ffe489e5a1af95253b153ce9d6f933e5dbabe14c494631234697a0e08'
+
+if ::Gem::Version.new(node['android-sdk']['version']) < ::Gem::Version.new('25')
```
Comment (rjaros87) [inline attributes/default.rb]: We check the SDK version and set `node['android-sdk']['legacy_sdk']` attribute which will be used in our recipies. Also there are different urls.

File: recipes/unix.rb [hunk 5]
```diff
@@ -71,6 +73,7 @@
   owner node['android-sdk']['owner']
   group node['android-sdk']['group']
   backup node['android-sdk']['backup_archive']
+  strip_components node['android-sdk']['legacy_sdk'] ? 1 : 0
```
Comment (rjaros87) [inline recipes/unix.rb]: Major fix for unix system is just simply extract archive to `android-sdk` directory without strip directory (by default ark enter to the first directory and then extract the content (default value: `1`))

File: recipes/windows.rb [hunk 6]
```diff
@@ -43,9 +43,15 @@
 
 ruby_block 'Rename directory' do
   block do
-    ::File.rename "#{setup_root}/android-sdk-windows", "#{setup_root}/#{node['android-sdk']['name']}"
+    if node['android-sdk']['legacy_sdk']
+      ::File.rename "#{setup_root}/android-sdk-windows", "#{setup_root}/#{node['android-sdk']['name']}"
+    else
```
Comment (rjaros87) [inline recipes/windows.rb]: Major fix for windows system

**Comment by rjaros87** [comment]
Extras are only `Rakefile`, `Gemfile` and `.rubocop`

`.kitchen.yml` Had a bad indentation and strangely combined attributes. Of course, I had to add tests for SDK 25 and SDK 24 

`.travis.yml` I had problem with Ubuntu machine, because they don't see package urls `404` - `apt-get update` solves the problem and by the way I raised versions of ruby to be the same as is in the ChefDK

`Berksfile` (test-kitchen) the same problems as was on Travis

`default.rb` and recipies - see my comments above

`test/integration/*` - because now we have two suites (default and legacy) I used shared context to not duplicate scenario :)

`dna.json` - just update sdk version

Hope my comments will help.

File: Rakefile (hunk 1 above)
Comment (rjaros87) [reply Rakefile:13]: and also we execute foodcritic ;)

**Comment by gildegoma** [comment]
Thank you *a lot* and *again* @rjaros87 and @Brantone to collaborate on this issue.
I'd like to precise that gildegoma#34 is a bug fix issue for the `0.2.x` version of the cookbook, and its fix must absolutely be *focused* and be based on cb67ebd25d656f47fb581ed9a81c9d11c50abb86 (`0.2.0` without any additional changes). The backwards compatible approach of auto-detected `node['android-sdk']['legacy_sdk']` looks great at first glance. I'd really love to receive a PR that only address the issue for the Linux platform (instead of being based on https://github.com/gildegoma/chef-android-sdk/pull/33 stuff). The work for version `0.3.0` is a quite larger topic, where more refactorings are fully allowed.

❤️ ❤️ ❤️

**Comment by rjaros87** [comment]
Merged with upstream master :) and refactor unix recipe

File: recipes/unix.rb [hunk 7]
```diff
@@ -71,17 +73,10 @@
...
   group node['android-sdk']['group']
   backup node['android-sdk']['backup_archive']
+  strip_components 0
   action node['android-sdk']['with_symlink'] ? :install : :put
 end
 
-#
-# Fix non-friendly 0750 permissions in order to make android-sdk available to all system users
-#
-%w(add-ons platforms tools).each do |subfolder|
```
Comment (rjaros87) [inline recipes/unix.rb:80]: we don't need this anymore :) by default in archive `tools` has `755`

**Pull request merged**